src/core/request_handler.rs

```diff
@@ -1,27 +1,19 @@
 // Copyright 2020 Contributors to the Parsec project.
 // SPDX-License-Identifier: Apache-2.0
-#![allow(dead_code)]
-
+use super::IpcHandler;
 use crate::error::{ClientErrorKind, Result};
 #[cfg(test)]
 use mockstream::SyncMockStream;
 use parsec_interface::requests::{Request, Response};
-use std::io::{Read, Write};
-#[cfg(not(test))]
-use std::os::unix::net::UnixStream;
-use std::path::PathBuf;
-use std::time::Duration;
 
 const DEFAULT_MAX_BODY_SIZE: usize = usize::max_value();
-const DEFAULT_SOCKET_PATH: &str = "/tmp/security-daemon-socket";
 
 /// Low level client structure to send a `Request` and get a `Response`.
 #[derive(Clone)]
 #[cfg_attr(not(test), derive(Debug))]
 pub struct RequestHandler {
     pub max_body_size: usize,
-    pub timeout: Option<Duration>,
-    pub socket_path: PathBuf,
+    pub ipc_handler: IpcHandler,
```

Comment:
This could contain instead:
```rust
pub speaker: Box<dyn Speak>
```
`Speak` to match `Listen` in the service 😉

Reply:
What do you think the benefit of this would be over `enum`?

Reply:
I think that would allow to remove the `[cfg(test)]` of "common" code, putting all test related code in its own module and in the future, be able to add more IPC handler without having to modify common code.

Reply:
> be able to add more IPC handler without having to modify common code. 

Not sure I follow

Reply:
Let me try to reformulate: with your current implementation, if you add a new IPC type handler, you will have to modify the `IpcHandler` enum (common code) to add it and add to the `get_stream` function (common code) the specific things to do for that IPC.
With using trait object, the common code can just define a trait and a trait object which means that current IPC type handlers (Unix socket and Mock stream) and the new ones can be written in isolation in their own module.

Reply:
I see what you mean, but I don't really see a clear benefit in not changing "common code".

The big tradeoff that I see is whether we want other people to be able to implement their own IPC mechanism (i.e. outside of the crate). I think it makes sense towards that goal 🤔

Reply:
Maybe that one benefit is a bit more dubious but what about the one removing test code (mock) from source code? That would definitely make it easier to read/maintain if those things are separated don't you think?

Reply:
Yeah, I think that makes sense! I'll probs create a `testing` module in `core` to put everything test-related there.

Reply:
or in `tests/` 🙊 (I don't mind do as you want)